### PR TITLE
fix(lazyWebpack): implement more proxy traps

### PR DESCRIPTION
**Pull request by DiamondMiner88**

### CODE REVIEW

File: src/utils/misc.tsx [hunk 1]
```diff
@@ -14,15 +14,19 @@ export function lazy<T>(factory: () => T): () => T {
...
-    return new Proxy({}, {
+    return new Proxy(() => null, {
         get: (_, prop) => getMod()[prop],
-        set: (_, prop, v) => getMod()[prop] = v
+        set: (_, prop, value) => getMod()[prop] = value,
+        has: (_, prop) => prop in getMod(),
+        apply: (_, $this, args) => (getMod() as Function).apply($this, args),
+        construct: (_, args, newTarget) => newTarget(args),
+        deleteProperty: (_, prop) => delete getMod()[prop],
+        defineProperty: (_, property, attributes) => !!Object.defineProperty(getMod(), property, attributes)
```
Comment (Vendicated) [inline src/utils/misc.tsx:29]: why the `!!`?

File: src/utils/misc.tsx [hunk 2]
```diff
@@ -14,15 +14,19 @@ export function lazy<T>(factory: () => T): () => T {
...
     const getMod = lazy(() => find(filter));
 
-    return new Proxy({}, {
+    return new Proxy(() => null, {
         get: (_, prop) => getMod()[prop],
-        set: (_, prop, v) => getMod()[prop] = v
+        set: (_, prop, value) => getMod()[prop] = value,
+        has: (_, prop) => prop in getMod(),
+        apply: (_, $this, args) => (getMod() as Function).apply($this, args),
+        construct: (_, args, newTarget) => newTarget(args),
```
Comment (Vendicated) [inline src/utils/misc.tsx:27]: shouldn't this use the `new` keyword?
```suggestion
        construct: (_, args, newTarget) => new newTarget(...args),
```

File: src/utils/misc.tsx [hunk 3]
```diff
@@ -14,15 +14,19 @@ export function lazy<T>(factory: () => T): () => T {
 /**
  * Do a lazy webpack search. Searches the module on first property access
  * @param filter Filter function
- * @returns Proxy. Note that only get and set are implemented, all other operations will have unexpected
- *          results.
+ * @returns A proxy to the webpack module. Not all traps are implemented, may produce unexpected results.
```
Comment (Vendicated) [inline src/utils/misc.tsx:17]: which traps are missing?

File: src/utils/misc.tsx (hunk 1 above)
Comment (DiamondMiner88) [reply src/utils/misc.tsx:29]: return type is boolean

File: src/utils/misc.tsx (hunk 3 above)
Comment (DiamondMiner88) [reply src/utils/misc.tsx:17]: getOwnPropertyDescriptor, getPrototypeOf, isExtensible, ownKeys, preventExtensions, setPrototypeOf

**Review by Vendicated — APPROVED** [review]

**Pull request merged**